db/migrations/20220509002323-create-github-app.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		return queryInterface.createTable('GithubApps', {
+			id: {
+				allowNull: false,
+				autoIncrement: true,
+				primaryKey: true,
+				type: Sequelize.INTEGER
+			},
+			githubBaseUrl: {
+				allowNull: false,
+				type: Sequelize.STRING
+			},
+			githubApiBaseUrl: {
+				allowNull: false,
+				type: Sequelize.STRING
+			},
+			githubServerType: {
+				allowNull: false,
+				type: Sequelize.ENUM("cloud", "ghe")
+
+			},
+			githubClientId: {
+				allowNull: false,
+				type: Sequelize.STRING
+			},
+			secrets: {
+				allowNull: false,
+				type: Sequelize.BLOB
```

Comment:
githubClientSecret, webhookSecret and privateKey will be stored in secrets vault.

Reply:
What do you mean with "secrets vault"? Does that mean they will be stored in this `secrets` blob, but encrypted?

Reply:
Blob is non-encrypted.  DB is encrypted at rest.  We need to be careful how we name things since we have Github Client Secret and Github Webhook Secret.  We also need to go over how we'll be listening to webhooks for each github enterprise instance as we need listening code for each one or have a way to decipher every event by iterating through all  github instances to see which webhook secret will work with the event, which I'm not sure if actually scalable.

We need to brainstorm some solutions for this.

Reply:
We need to store githubClientSecret, webhookSecret and privateKey as encrypted values. The idea here is to put all these values in a secure vault. The vault will be stored as blob in secrets column. The same pattern is being used in installations table with sequelize-encrypted library.

Reply:
ah right, didn't know that was the idea without he model being set